apps/prairielearn/assets/scripts/lib/questionsTable.js

```diff
@@ -0,0 +1,194 @@
+/* eslint-env browser, jquery */
+
+import _ from 'lodash';
+import { onDocumentReady, decodeData } from '@prairielearn/browser-utils';
+import { html } from '@prairielearn/html';
+
+onDocumentReady(() => {
+  const { course_instance_ids, showAddQuestionButton, urlPrefix, plainUrlPrefix } =
+    decodeData('questions-table-data');
+  window.topicList = function () {
+    var data = $('#questionsTable').bootstrapTable('getData');
+    return _.keyBy(_.map(data, (row) => row.topic.name));
+  };
+
+  window.tagsList = function () {
+    var data = $('#questionsTable').bootstrapTable('getData');
+    return _.keyBy(_.map(_.flatten(_.filter(_.map(data, (row) => row.tags))), (row) => row.name));
+  };
+
+  window.versionList = function () {
+    var data = $('#questionsTable').bootstrapTable('getData');
+    return _.keyBy(_.map(data, (row) => row.display_type));
+  };
+
+  window.qidFormatter = function (qid, question) {
+    var text = '';
+    if (question.sync_errors) {
```

Comment:
When we reuse this component for viewing shared questions we'll need an option to hide sync errors and other things that are only relevant for the owner of the questions. I'm fine to leave that for the later PR that adds shared-question viewing.

Reply:
ya, I'll worry about that sort of thing next PR